### PR TITLE
Modern Apartment JSONify

### PR DESCRIPTION
**Pull request by vickio**
This is a conversion of the modern apartment tower from `map::draw_map` to JSON. It handles save file conversion for the renamed overmap tiles.

I only made a couple of minor changes:
- added curtains to the windows (with the curtains randomly drawn)
- interior doors may randomly be open

### CODE REVIEW

**Comment by codemime** [comment]
I'll test this soon. Work on jsonizing the mapgen spaghetti is very welcome.

**Comment by codemime** [comment]
`mapgen_apartments_mod_tower_1_entrance()` is still in the code. 
It needs to be removed along with the stub for `mapgen_apartments_mod_tower_1`.

File: src/savegame.cpp [hunk 1]
```diff
@@ -334,28 +335,31 @@ void overmap::convert_terrain( const std::unordered_map<tripoint, std::string> &
...
-            nearby.push_back( { 1, other, -1, other, "apartments_con_tower_NE_south" } );
-            nearby.push_back( { -1, other, -1, other, "apartments_con_tower_NE_east" } );
-            nearby.push_back( { 1, other, 1, other, "apartments_con_tower_NE_west" } );
-            nearby.push_back( { -1, entr, -1, other, "apartments_con_tower_SE_north" } );
-            nearby.push_back( { 1, entr, 1, other, "apartments_con_tower_SE_south" } );
-            nearby.push_back( { 1, other, -1, entr, "apartments_con_tower_SE_east" } );
-            nearby.push_back( { -1, other, 1, entr, "apartments_con_tower_SE_west" } );
+        if( old == "apartments_con_tower_1_entrance" ||
+            old == "apartments_mod_tower_1_entrance" ) {
+            const std::string base = old.substr( 0, 21 );
```
Comment (codemime) [inline src/savegame.cpp]: This `substr()` is quite fragile due to the magic number.

File: src/savegame.cpp [hunk 2]
```diff
@@ -334,28 +335,31 @@ void overmap::convert_terrain( const std::unordered_map<tripoint, std::string> &
...
+            const std::string other = base + "1";
+            nearby.push_back( { 1, other, -1, other, base + "SW_north" } );
+            nearby.push_back( { -1, other, 1, other, base + "SW_south" } );
+            nearby.push_back( { 1, other, 1, other, base + "SW_east" } );
+            nearby.push_back( { -1, other, -1, other , base + "SW_west" } );
+
+        } else if( old == "apartments_con_tower_1" || "apartments_mod_tower_1" ) {
```
Comment (codemime) [inline src/savegame.cpp]: This condition is broken: it always evaluates to `true`.

**Comment by vickio** [comment]
Sorry about that, should be better now.

File: data/json/overmap_specials.json [hunk 3]
```diff
@@ -480,14 +480,14 @@
         "type" : "overmap_special",
         "id" : "Apartments_Mod",
         "overmaps" : [
-            { "point":[0,0,0], "overmap": "apartments_mod_tower_1"},
-            { "point":[1,0,0], "overmap": "apartments_mod_tower_1_entrance"},
-            { "point":[0,1,0], "overmap": "apartments_mod_tower_1"},
-            { "point":[1,1,0], "overmap": "apartments_mod_tower_1"}
+            { "point": [ 0, 0, 0 ], "overmap": "apartments_mod_tower_NW_north" },
+            { "point": [ 1, 0, 0 ], "overmap": "apartments_mod_tower_NE_north" },
+            { "point": [ 0, 1, 0 ], "overmap": "apartments_mod_tower_SW_north", "connect": "road" },
```
Comment (codemime) [inline data/json/overmap_specials.json]: `"connect": "road"` isn't supported anymore. Only `"connections" : [ ... ]` is used.

**Comment by codemime** [comment]
> Sorry about that

There's no need to be sorry. Good job. I hope there will be more PRs like this.

**Pull request merged**